native/src/main/java/io/ballerina/stdlib/oracledb/parameterprocessor/OracleDBResultParameterProcessor.java

```diff
@@ -158,19 +159,19 @@ public Object processCustomTypeFromResultSet(ResultSet resultSet, int columnInde
             case OracleTypes.TIMESTAMPLTZ:
                 return processTimestampWithTimezoneResult(resultSet, columnIndex, sqlType, ballerinaType);
             default:
-                throw new ApplicationError("Unsupported SQL type " + columnDefinition.getSqlName());
+                throw new DataError("Unsupported SQL type " + columnDefinition.getSqlName());
         }
     }
 
     @Override
     public Object processCustomOutParameters(
-            CallableStatement statement, int paramIndex, int sqlType) throws ApplicationError {
+            CallableStatement statement, int paramIndex, int sqlType) throws DataError {
         switch (sqlType) {
             case OracleTypes.INTERVALDS:
             case OracleTypes.INTERVALYM:
                 return processInterval(statement, paramIndex);
             default:
-                throw new ApplicationError("Unsupported SQL type '" + sqlType + "' when reading Procedure call " +
+                throw new DataError("Unsupported SQL type '" + sqlType + "' when reading Procedure call " +
```

Comment:
Shall we use string template to create the error message ? Check for other places as well.

Reply:
corrected in https://github.com/ballerina-platform/module-ballerinax-oracledb/pull/25/commits/57e476436f96c585352e768a7e39b8988773c1ed